src/app/subnets/views/SubnetsList/SubnetsTable/components.tsx

```diff
@@ -59,3 +70,143 @@ export const CellContents = ({
+      key: SubnetsColumns.DHCP,
+      content: content[SubnetsColumns.DHCP],
+    },
+    {
+      "aria-label": SubnetsColumns.SUBNET,
+      key: SubnetsColumns.SUBNET,
+      content: content[SubnetsColumns.SUBNET],
+    },
+    {
+      "aria-label": SubnetsColumns.IPS,
+      key: SubnetsColumns.IPS,
+      content: content[SubnetsColumns.IPS],
+    },
+    {
+      "aria-label": SubnetsColumns.SPACE,
+      key: SubnetsColumns.SPACE,
+      content: content[SubnetsColumns.SPACE],
+      className: "u-align--right",
+    },
+  ];
+
+  return {
+    key,
+    className: classNames(classes),
+    columns,
+  };
+};
+
+export const generateSubnetRows = (subnets: SubnetsTableRow[]) => {
+  return subnets.map((subnet, index) => {
+    const content = {
+      [SubnetsColumns.FABRIC]: (
+        <CellContents value={subnet[SubnetsColumns.FABRIC]} />
+      ),
+      [SubnetsColumns.VLAN]: (
+        <CellContents value={subnet[SubnetsColumns.VLAN]} />
+      ),
+      [SubnetsColumns.DHCP]: (
+        <CellContents value={subnet[SubnetsColumns.DHCP]} />
+      ),
+      [SubnetsColumns.SUBNET]: (
+        <CellContents value={subnet[SubnetsColumns.SUBNET]} />
+      ),
+      [SubnetsColumns.IPS]: <CellContents value={subnet[SubnetsColumns.IPS]} />,
+      [SubnetsColumns.SPACE]: (
+        <CellContents value={subnet[SubnetsColumns.SPACE]} />
+      ),
+    };
+    return generateSubnetRow({
+      key: `${subnet.sortData.vlanId}-${subnet.sortData.fabricId}-${index}`,
+      content,
+      classes: "subnet-row",
+    });
+  });
+};
+
+export const generateSubnetGroupRows = ({
+  itemName,
+  groups,
+  columnLength,
+  groupMap,
+}: {
+  itemName: string;
+  groups: FabricTableRow[] | SpaceTableRow[];
+  columnLength: number;
+  groupMap: Record<
+    string | number,
+    {
+      count: number;
+    }
+  >;
+}) => {
+  let rows: MainTableRow[] = [];
+  groups.forEach((group) => {
+    const { networks } = group;
+    if ("fabricName" in group) {
+      const { fabricName } = group;
+      rows.push({
+        "aria-label": `${fabricName} group`,
+        className: "",
+        columns: [
+          {
+            colSpan: columnLength,
+            content: (
+              <GroupRow
+                count={groupMap[fabricName].count}
+                groupName={`${fabricName}`}
+                itemName={itemName}
+              />
+            ),
+          },
+        ],
+      });
+    } else {
+      const { spaceName } = group;
+      rows.push({
+        "aria-label": `${spaceName} group`,
+        className: "",
+        columns: [
+          {
+            colSpan: columnLength,
+            content: (
+              <GroupRow
+                count={groupMap[spaceName].count}
+                groupName={`${spaceName}`}
+                itemName={itemName}
+              />
+            ),
+          },
+        ],
+      });
+    }
+
+    rows = rows.concat(generateSubnetRows(networks));
+  });
+  return rows;
+};
```

Comment:
There'a some repetition here that could be avoided which would also make this easier to understand. One way to do that below.


```suggestion
export const generateSubnetGroupRows = ({
  itemName,
  groups,
  columnLength,
  groupMap,
}: {
  itemName: string;
  groups: FabricTableRow[] | SpaceTableRow[];
  columnLength: number;
  groupMap: Record<string | number, { count: number }>;
}) => {
  const generateGroupRow = (name: string) => {
    return {
      "aria-label": `${name} group`,
      className: "",
      columns: [
        {
          colSpan: columnLength,
          content: (
            <GroupRow
              count={groupMap[name].count}
              groupName={name}
              itemName={itemName}
            />
          ),
        },
      ],
    };
  };

  return groups.flatMap((group) => {
    const { networks } = group;
    const name = "fabricName" in group ? group.fabricName : group.spaceName;
    return [generateGroupRow(name as string), ...generateSubnetRows(networks)];
  });
};
```